Gems/AWSNativeSDKInit/Code/Source/Clients/AWSMemoryInterface.cpp

```diff
@@ -22,14 +23,14 @@ namespace AWSNativeSDKInit
     {
     }
 
-    void* MemoryManager::AllocateMemory(std::size_t blockSize, std::size_t alignment, const char* allocationTag) 
+    void* MemoryManager::AllocateMemory(std::size_t blockSize, std::size_t alignment, [[maybe_unused]]const char* allocationTag) 
```

Comment:
`allocationTag` isn't used anymore? Maybe just remove the parameter here and in the header.

Reply:
The method is an override to Aws::Utils::Memory::MemorySystemInterface. We just can't do anything with it on O3DE side :/

Reply:
Got it. I plan on writing an RFC for recommendations on how to handle these situations, because it comes up a lot, and there are currently at least 3 ways they are handled. Once the RFC is ratified, i'll update the O3DE coding standards. But for situations like these, my opinion is that [[maybe_unused]] should only be used for cases where a variable is conditionally compiled out due to an ifdef like `if defined(RELEASE)` which is the case for all the logging macros, since logging is disabled for release builds. Some like above I'd recommend (but you don't have to do it)
```suggestion
    void* MemoryManager::AllocateMemory(std::size_t blockSize, std::size_t alignment, const char* /*allocationTag*/) 
```

Reply:
fixed